ContactApp3/View Controllers/ContactsListViewModel.swift

```diff
@@ -54,6 +56,66 @@ class ContactsListViewModel {
             print("Error")
             
         }
+    }
+    
+    func saveContact(contact: Person) {
+        // Get access to the contacts store
+        let store = CNContactStore()
+        
+        // Create a mutable object to store data in contactsStore(not thread-safe class), CNContacts is a(thread-safe class) but provide a get only property
+        let contacts = CNMutableContact()
+        if let contactName = contact.name {
+              contacts.givenName = contactName
+           }
+        
+        // Store the phone number in phoneNumbers dictionary with label(main phone)
+        if let contactNumber = contact.number {
+        contacts.phoneNumbers = [CNLabeledValue(label: CNLabelPhoneNumberMain, value: CNPhoneNumber(stringValue: contactNumber))]
+            }
+        
+        // Store the profile picture as data
+        if let contactImage = contact.image {
+            contacts.imageData = contactImage.jpegData(compressionQuality: 1.0)
+        }
+        
+        // Save
+        let saveRequest = CNSaveRequest()
+        saveRequest.add(contacts, toContainerWithIdentifier: nil)
+        try? store.execute(saveRequest)
+    }
+    
+    func deleteContact(contact: Person) {
+        
+        // Get access to the contacts store
+        let store = CNContactStore()
+        
+        //Specify which data keys we want to fetch
+        let keys = [CNContactGivenNameKey, CNContactFamilyNameKey, CNContactPhoneNumbersKey, CNContactImageDataKey] as [CNKeyDescriptor]
+        
+        //Create fetch request
+        let fetchRequest = CNContactFetchRequest(keysToFetch: keys)
+        
+        //Call method to fetch all contacts
+        do {
+            try store.enumerateContacts(with: fetchRequest, usingBlock: { toDeleteContact, result in
+                if contact.name == toDeleteContact.givenName + " " + toDeleteContact.familyName {
+                    // Save
+                    let saveRequest = CNSaveRequest()
+                    saveRequest.delete(toDeleteContact.mutableCopy() as! CNMutableContact)
```

Comment:
@oulamardawi don't use force casting "as!" instead use optional casting "as?" to avoid crashes